File: client/mc/1.21.1-neoforge/src/main/java/calebxzhou/rdi/mc/client/rcmd/RcmdClientCommands.java
package calebxzhou.rdi.mc.client.rcmd;

import calebxzhou.rdi.mc.rcmd.RcmdDispatchResult;
import calebxzhou.rdi.mc.rcmd.RcmdResult;
import net.minecraft.client.Minecraft;

public final class RcmdClientCommands {
    private RcmdClientCommands() {
    }

    public static boolean isRcmd(String message) {
        return message != null && message.startsWith("\\");
    }

    public static RcmdDispatchResult dispatch(Minecraft minecraft, String message) {
        return calebxzhou.rdi.mc.common2.rcmd.client.RcmdClientCommands.dispatch(new RcmdClientBridge211(minecraft), message);
    }

    public static void reply(Minecraft minecraft, RcmdResult result) {
        calebxzhou.rdi.mc.common2.rcmd.client.RcmdClientCommands.reply(new RcmdClientBridge211(minecraft), result);
    }
}

File: client/mc/1.21.1-neoforge/src/main/kotlin/calebxzhou/rdi/mc/client/rcmd/RcmdClientCommands.kt
package calebxzhou.rdi.mc.client.rcmd

import calebxzhou.rdi.mc.rcmd.RcmdDispatchResult
import calebxzhou.rdi.mc.rcmd.RcmdResult
import net.minecraft.client.Minecraft

object RcmdClientCommands {
    @JvmStatic
    fun isRcmd(message: String?): Boolean {
        return message != null && message.startsWith("\\")
    }

    fun dispatch(minecraft: Minecraft, message: String): RcmdDispatchResult {
        return calebxzhou.rdi.mc.common2.rcmd.client.RcmdClientCommands.dispatch(RcmdClientBridge211(minecraft), message)
    }

    fun reply(minecraft: Minecraft, result: RcmdResult) {
        calebxzhou.rdi.mc.common2.rcmd.client.RcmdClientCommands.reply(RcmdClientBridge211(minecraft), result)
    }
}
